pkg/clusteragent/admission/mutate/autoinstrumentation/auto_instrumentation_test.go

```diff
@@ -37,6 +37,169 @@ import (
+			},
+		},
+		{
+			name:                  "nominal case: java & python",
+			pod:                   common.FakePod("java-pod"),
+			expectedInjectorImage: commonRegistry + "/apm-injector:v27",
+			libsToInject: []libInfo{
+				java.libInfo("", "gcr.io/datadoghq/dd-lib-java-init:v1"),
+				python.libInfo("", "gcr.io/datadoghq/dd-lib-python-init:v1"),
+			},
+		},
+		{
+			name: "java + injector-tag-override",
+			pod: common.FakePodSpec{
+				Annotations: map[string]string{
+					"admission.datadoghq.com/apm-injector.version": "v0",
+				},
+			}.Create(),
+			expectedInjectorImage: commonRegistry + "/apm-injector:v0",
+			libsToInject: []libInfo{
+				java.libInfo("", "gcr.io/datadoghq/dd-lib-java-init:v1"),
+			},
+		},
+		{
+			name: "java + injector-image-override",
+			pod: common.FakePodSpec{
+				Annotations: map[string]string{
+					"admission.datadoghq.com/apm-injector.custom-image": "docker.io/library/apm-inject-package:v27",
+				},
+			}.Create(),
+			expectedInjectorImage: "docker.io/library/apm-inject-package:v27",
+			libsToInject: []libInfo{
+				java.libInfo("", "gcr.io/datadoghq/dd-lib-java-init:v1"),
+			},
+		},
+	}
+
+	wmeta := fxutil.Test[workloadmeta.Component](
+		t,
+		core.MockBundle(),
+		workloadmetafxmock.MockModule(),
+		fx.Supply(workloadmeta.NewParams()),
+	)
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			c := configmock.New(t)
+
+			c.SetWithoutSource("apm_config.instrumentation.version", "v2")
+
+			webhook := mustWebhook(t, wmeta)
+			require.Equal(t, instrumentationV2, webhook.version)
+			require.True(t, webhook.version.usesInjector())
+
+			err := webhook.injectAutoInstruConfig(tt.pod, tt.libsToInject, false, "")
+			if tt.wantErr {
+				require.Error(t, err, "expected injectAutoInstruConfig to error")
+			} else {
+				require.NoError(t, err, "expected injectAutoInstruConfig to succeed")
+			}
+
+			if err != nil {
+				return
+			}
+
+			envsByName := map[string]corev1.EnvVar{}
+			for _, env := range tt.pod.Spec.Containers[0].Env {
+				envsByName[env.Name] = env
+			}
+
+			requireEnv := func(t *testing.T, key string, ok bool, value string) {
+				t.Helper()
+				val, exists := envsByName[key]
+				require.Equal(t, ok, exists, "expected env %v to be %v", key, ok)
+				require.Equal(t, value, val.Value, "expected env %v = %v", key, val)
+			}
+
+			for _, env := range injectAllEnvs() {
+				requireEnv(t, env.Name, false, "")
+			}
+
+			if len(tt.libsToInject) == 0 {
+				require.Zero(t, len(tt.pod.Spec.InitContainers), "no libs, no init containers")
+				requireEnv(t, "LD_PRELOAD", false, "")
+			} else {
+
+				require.Equal(t, volumeName, tt.pod.Spec.Volumes[0].Name,
+					"expected datadog volume to be injected")
+
+				for i, c := range tt.pod.Spec.InitContainers {
+
+					var injectorMountPath string
+
+					if i == 0 { // check init container
+						require.Equal(t, "datadog-init-apm-inject", c.Name,
+							"the first init container is the injection container")
```

Comment:
nit: this is an error msg, so I think it should be "...init container should be the..." to know what failed when we see this error.

There are other cases like this one in this test.